src/rules/fileHeaderRule.ts

```diff
@@ -63,7 +71,9 @@ export class Rule extends Lint.Rules.AbstractRule {
             offset,
             (pos, end, kind) => text.substring(pos + 2, kind === ts.SyntaxKind.SingleLineCommentTrivia ? end : end - 2));
 
-        if (commentText === undefined || !headerFormat.test(commentText)) {
+        const trailingNewLineViolation = enforceExtraTrailingLine ? this.doesNewLineEndingViolationExist(text, offset) : false;
```

Comment:
I think this config option is only relevant when the file header exists and is otherwise valid. So:
```ts
const trailingNewLineViolation = 
    enforceExtraTrailingLine &&
    headerFormat.test(commentText!) &&
    this.doesNewLineEndingViolationExist(text, offset)
```

In fact, since we need to a return a `RuleFailure` when `trailingNewLineViolation` is `true`, you could just inline this expression in an `if` statement;